packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -443,24 +456,48 @@ export class Nodegroup extends Resource implements INodegroup {
   }
 }
 
-function getAmiTypeForInstanceType(instanceType: InstanceType) {
-  return INSTANCE_TYPES.graviton2.includes(instanceType.toString().substring(0, 3)) ? NodegroupAmiType.AL2_ARM_64 :
-    INSTANCE_TYPES.graviton.includes(instanceType.toString().substring(0, 2)) ? NodegroupAmiType.AL2_ARM_64 :
-      INSTANCE_TYPES.gpu.includes(instanceType.toString().substring(0, 2)) ? NodegroupAmiType.AL2_X86_64_GPU :
-        INSTANCE_TYPES.inferentia.includes(instanceType.toString().substring(0, 4)) ? NodegroupAmiType.AL2_X86_64_GPU :
-          NodegroupAmiType.AL2_X86_64;
+/**
+ * AMI types of different architectures. Make sure AL2 is always the first element, which will be the default AmiType if amiType and
+ * launchTemplateSpec are both undefined.
+ */
+const arm64AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_ARM_64, NodegroupAmiType.BOTTLEROCKET_ARM_64];
+const x8664AmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64, NodegroupAmiType.BOTTLEROCKET_X86_64];
+const gpuAmiTypes: NodegroupAmiType[] = [NodegroupAmiType.AL2_X86_64_GPU];
+
+/**
+ * This function returns an array of "possible AMI types" from the given instance type
+ * @param instanceType The instance type
+ * @returns NodegroupAmiType[]
+ */
+function getPossibleAmiTypesForInstanceType(instanceType: InstanceType): NodegroupAmiType[] {
+  return INSTANCE_TYPES.graviton2.includes(instanceType.toString().substring(0, 3)) ? arm64AmiTypes :
+    INSTANCE_TYPES.graviton.includes(instanceType.toString().substring(0, 2)) ? arm64AmiTypes :
+      INSTANCE_TYPES.gpu.includes(instanceType.toString().substring(0, 2)) ? gpuAmiTypes :
+        INSTANCE_TYPES.inferentia.includes(instanceType.toString().substring(0, 4)) ? gpuAmiTypes :
+          x8664AmiTypes;
 }
 
 // this function examines the CPU architecture of every instance type and determines
-// what ami type is compatible for all of them. it either throws or produces a single value because
+// what ami type is compatible for all of them. it either throws or produces an array of possible instance types because
 // instance types of different CPU architectures are not supported.
-function getAmiType(instanceTypes: InstanceType[]) {
-  const amiTypes = new Set(instanceTypes.map(i => getAmiTypeForInstanceType(i)));
-  if (amiTypes.size == 0) { // protective code, the current implementation will never result in this.
+function getPossibleAmiTypes(instanceTypes: InstanceType[]): NodegroupAmiType[] {
+  const amiTypes = new Set<NodegroupAmiType>();
+  for (const t of instanceTypes) {
+    getPossibleAmiTypesForInstanceType(t).forEach(x => amiTypes.add(x));
+  }
+  if (new Set(amiTypes).size == 0) { // protective code, the current implementation will never result in this.
```

Comment:
1. use `===`
2. don't think we need to construct a new set here

```suggestion
  if (amiTypes.size === 0) { // protective code, the current implementation will never result in this.
```